[20기 문성호]마이페이지 구현

Room9 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- my page 구현 
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
-decorator를 통해 인가 받기
- 인가 후 스킨타입 / 주소 추가 및 삭제 가능

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
[1차 프로젝트] wesop 장바구니 1차commit

Commit:
ADD: Mypage 구현
로그인 상태에서 skintype과 address 수정 가능
decorator 인증과정 동작확인
skintype 추가/삭제, address 추가/삭제 확인완료

Commit:
refactor: order app 내용 삭제

Commit:
refactor: view 1차 수정

Commit:
refactor: view 1차 수정

Commit:
refactor: 피드백 수정

Commit:
refactor: 피드백 수정
- UserInformationView로 통일 및 url endpoint 수정
- post 시 요청값에  ''와 'empty'를 구분하여 skintype과 address 개별 변경 가능하도록 구현

Kang-kyunghun commented on users/urls.py:
```diff
@@ -1,8 +1,8 @@
 
 urlpatterns = [
-            path('/signup', SignupView.as_view())
+            path('/userinfo', UserInformationView.as_view()),
```
```suggestion
            path('/information', UserInformationView.as_view()),
```
main url.py에 user가 이미 있기 때문에 다음과 같이 수정하면 좋을 것 같습니다.

Kang-kyunghun commented on users/views.py:
```diff
@@ -4,58 +4,34 @@
+    def post(self, request):
+        data = json.loads(request.body)
+        user = request.user
+        info = 'Nothing'
```
infor라는 변수 명은 명확하지 않습니다. 해당 변수가 무엇을 의미하는지 정확히 나타낼 수 있도록 바꿔 주시고 information을 infor처럼 줄이지 말아주세요.

Room9 replied:
넵 수정완료했습니다

Kang-kyunghun commented on users/views.py:
```diff
@@ -4,58 +4,34 @@
+            if data['skin_type'] != '' :
+                info              = data['skin_type']
+                skin              = SkinType.objects.get(name=info) if info!='empty' else '' 
+                skin_id           = skin.id if skin else ''
```
skin_id = '' 가 되었을 때  user.skin_type_id = skin_id 이부분에서 에러가 날 것 같은데 확인 바랍니다.

Room9 replied:
skin_type = 'empty'로 요청했을 때, skin_id=''가 되는데 정상 구동 확인했습니다
![image](https://user-images.githubusercontent.com/79136484/118802507-28fec300-b8dd-11eb-9a28-f57b60270a68.png)

Kang-kyunghun commented on users/utils.py:
```diff
@@ -0,0 +1,37 @@
+import  jwt
```
데코레이터 부분만 따로 브렌치 만들어서 머지 받으면 좋을 것 같아요

Room9 replied:
따로 브랜치 만들어서 pr 올리겠습니다

Commit:
commit

Commit:
refactor: 피드백 수정
- 변수명과 url endpoint 명을 info -> information으로 변경했습니다
skin_type='empty'로 post 시 skin_id=''가 되는데 error 없이 skin_type_id가 null로 변경되는 것 확인했습니다

Commit:
wip

Commit:
refactor: 리뷰 사항 반영

Kang-kyunghun commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/Users/mac/miniconda3/envs/wesop/bin/python"
+}
```
성호님 계속 불피요한 파일들이 올라옵니다.

Room9 replied:
네넵 ignore 하겠습니다

Kang-kyunghun commented on users/views.py:
```diff
@@ -4,58 +4,34 @@
 
 from django.http  import JsonResponse
 from django.views import View
+from django.core.exceptions import ObjectDoesNotExist
 
 from users.models import User, SkinType
 from my_settings  import SECRET
 
+from users.utils  import Authorization_decorator
```
import 라인에 맞춰주세요.

Room9 replied:
넵 수정하겠습니다

Commit:
refactor: 리뷰사항 수정2

Commit:
conflict해결

Commit:
conflict해결

Kang-kyunghun commented:
LGTM~~!!!
